src/astar.cpp

```diff
@@ -630,45 +566,89 @@ ASR_RETVAL fpathAStarRoute(MOVE_CONTROL *psMove, PATHJOB *psJob)
 	{
 		// Copy the list, in reverse.
 		std::copy(path.rbegin(), path.rend(), psMove->asPath.data());
-
-		if (!context.isBlocked(tileOrig.x, tileOrig.y))  // If blocked, searching from tileDest to tileOrig wouldn't find the tileOrig tile.
-		{
-			// Next time, search starting from nearest reachable tile to the destination.
-			fpathInitContext(context, psJob->blockingMap, tileDest, pred.nearestCoord, tileOrig, dstIgnore);
-		}
```

Comment:
This may be behind some of the issue?, given how the lag is produced and given the initial temporary lag spikes started in https://github.com/Warzone2100/warzone2100/commit/a65e3f8234eea09fcc3af2e277cea1f3e2e086fc which changed the value that could be passed in (`pred.nearestCoord` after that commit does not always equal `context.nearestCoord` before that commit)